src/main/java/picard/sam/SamAlignmentMerger.java

```diff
@@ -158,6 +141,70 @@ public void mergeAlignment(final File referenceFasta) {
         }
     }
 
+    @Override
+    protected SAMSequenceDictionary getDictionaryForMergedBam() {
+        SAMSequenceDictionary finalDict = null;
+        if (alignedSamFile != null && !alignedSamFile.isEmpty()) {
+            for (final File alignedFile : alignedSamFile) {
+                final SAMSequenceDictionary dict = SAMSequenceDictionaryExtractor.extractDictionary(alignedFile);
+                if (finalDict == null) {
+                    finalDict = dict;
+                } else {
+                    if (!compareSequenceDictionaries(finalDict, dict)) {
+                        log.error("Dictionaries of input files differ. Unable to continue.");
+                        throw new UnsupportedOperationException("Cannot use input files with different dictionaries.");
+                    }
+                }
+            }
+        } else {
+            final SeparateEndAlignmentIterator mergingIterator = new SeparateEndAlignmentIterator(this.read1AlignedSamFile, this.read2AlignedSamFile, referenceFasta);
+            finalDict = mergingIterator.getHeader().getSequenceDictionary();
+        }
+
+        SAMSequenceDictionary referenceDict = SAMSequenceDictionaryExtractor.extractDictionary(referenceFasta);
+        if (referenceDict == null) {
+            throw new PicardException("No sequence dictionary found for " + referenceFasta.getAbsolutePath() +
+                    ".  Use CreateSequenceDictionary.jar to create a sequence dictionary.");
+        }
+        return mergeDictionaries(finalDict, SAMSequenceDictionaryExtractor.extractDictionary(referenceFasta));
+    }
+
+    private SAMSequenceDictionary mergeDictionaries(SAMSequenceDictionary dict1, SAMSequenceDictionary dict2) {
```

Comment:
Would it make sense to have this be a static method in SAMSequenceDictionary instead?